lib/beaker-http/http.rb

```diff
@@ -1,79 +1,90 @@
 module Beaker
   module Http
 
-    # Beaker Http objects are essentially wrappers for a connection object that
-    # utilizes a Beaker::Host object for easier setup during testing.
-    class Http
+    # == Beaker::Http::Connection object instantiation examples
+    # These examples are for using the Connection object directly. If you are trying to use
+    # this from within a test, consider using the
+    # {Beaker::DSL::Helpers::WebHelpers DSL constructors} instead.
+    # @see Beaker::DSL::Helpers::WebHelpers
+    class Connection
       include Beaker::Http::Helpers
       extend Forwardable
 
-      attr_accessor :connection, :host
+      attr_reader :connection
 
-      def initialize(host)
-        @host = host
-        if @host.is_a?(Beaker::Host)
-          @connection = create_default_connection_with_beaker_host
-        else
-          raise "Argument host must be Beaker::Host"
-        end
+      # Beaker::Http::Connection objects can be instantiated with object that
+      # utilizes a  object for easier setup during testing.
+      #
+      # @param [Hash] options Typically the global options provided by Beaker.
+      # @option options [Beaker::Logger] :logger
+      # @option options [Boolean] :log_http_bodies
+      def initialize(options)
+        @connection = create_default_connection(options)
       end
 
-      def_delegators :@connection, :get, :post, :put, :delete, :head, :path
+      def_delegators :connection, :get, :post, :put, :delete, :head, :patch, :url_prefix, :url_prefix=, :ssl, :options
 
-      def create_default_connection
+      def create_default_connection(options)
         Faraday.new do |conn|
           conn.request :json
 
           conn.response :follow_redirects
           conn.response :raise_error
           conn.response :json, :content_type => /\bjson$/
-          conn.response :faraday_beaker_logger, @host, bodies: true
+
+          # We can supply a third argument, a Hash with key of :bodies set to true or false,
+          # to configure whether or not to log http bodies in requests and responses.
+          # However, to uncomplicate things, we will just use the default
+          # set in the middleware and not allow the http log level to be set
+          # independently of the beaker log level. If we find that we should allow setting
+          # of http bodies independent of the beaker log level, we should expose that setting
+          # here.
```

Comment:
Patrick already updated my initial PR to Scooter because he wanted more granularity of logging bodies vs not logging bodies.  For HA, we're getting a few responses that are so large that they end up taking many mb of space just for the logs when we log bodies so we need to be able to turn that off.  I think this will need to be included before modifying Scooter to depend on this library.